public/js/PasswordScreen.js

```diff
@@ -1,95 +1,73 @@
 /* eslint-disable no-unused-vars */
 /* eslint-disable react/jsx-no-undef */
 /* eslint-disable no-undef */
-class PasswordScreen extends React.Component {
-  constructor(props) {
-    super(props);
 
-    this.state = {
-      showModal: true,
-      password: ''
-    };
+const PasswordScreen = props => {
+  const [password, setPassword] = useState('');
 
-    this.handleModalShow = this.handleModalShow.bind(this);
-    this.handleModalClose = this.handleModalClose.bind(this);
-    this.handleFormSubmit = this.handleFormSubmit.bind(this);
+  // Get Pass Screen translation:
+  const pass_screen = props.jsonStrings.pass_screen;
 
-    this.handleInputChange = this.handleInputChange.bind(this);
-  }
-
-  handleFormSubmit(e) {
+  const handleFormSubmit = e => {
+    console.log('okkkk', props.configuration);
     e.preventDefault();
 
-  }
-
-  handleModalShow() {
-    this.setState({
-      showModal: true
+    // Send password to the backend:
+    props.sendWebSocket('verify-password', {
+      password,
+      loginWindowMinutes:
+        props.configuration.botOptions.login.loginWindowMinutes
     });
-  }
+  };
 
-  handleModalClose() {
-    this.setState({
-      showModal: false
-    });
-  }
-
-  handleInputChange(event) {
+  /**
+   * Update the password in state.
+   * @param {*} event
+   */
+  const handleInputChange = event => {
     const value = event.target.value;
-    this.state.password = value;
+    setPassword(value);
+  };
 
-    const { configuration } = this.props;
-    const typedPassword = {
-      pass: Array.from(value),
-      config: configuration
-    };
-    this.props.sendWebSocket('verify-password', {
-      typedPassword
-    });
+  if (_.isEmpty(props.jsonStrings)) {
```

Comment:
You can use destructuring here.

Reply:
> You can use destructuring here.

Mmmh ok but i dont know how here 🤔. Ia havnt idea for the best way to do..

Reply:
Like this:

`const { jsonStrings: { pass_screen } } = this.props;`